src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicForwardCancellationTokenToInvocationsFixer.TypeNameVisitor.vb

```diff
@@ -0,0 +1,311 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
+    Partial Public NotInheritable Class BasicForwardCancellationTokenToInvocationsFixer
+        Private Class TypeNameVisitor
+            Inherits SymbolVisitor(Of TypeSyntax)
+
+            Public Shared Function GetTypeSyntaxForSymbol(symbol As INamespaceOrTypeSymbol) As TypeSyntax
+                Return symbol.Accept(New TypeNameVisitor()).WithAdditionalAnnotations(Simplifier.Annotation)
+            End Function
+
+            Public Overrides Function DefaultVisit(symbol As ISymbol) As TypeSyntax
+                Throw New NotImplementedException()
+            End Function
+
+            Public Overrides Function VisitAlias(symbol As IAliasSymbol) As TypeSyntax
+                Return AddInformationTo(ToIdentifierName(symbol.Name))
+            End Function
+
+            Public Overrides Function VisitArrayType(symbol As IArrayTypeSymbol) As TypeSyntax
+                Dim underlyingNonArrayType = symbol.ElementType
+                While underlyingNonArrayType.Kind = SymbolKind.ArrayType
+                    underlyingNonArrayType = DirectCast(underlyingNonArrayType, IArrayTypeSymbol).ElementType
+                End While
+
+                Dim elementTypeSyntax = underlyingNonArrayType.Accept(Me)
+                Dim ranks = New List(Of ArrayRankSpecifierSyntax)()
+                Dim arrayType = symbol
+                While arrayType IsNot Nothing
+                    Dim commaCount = Math.Max(0, arrayType.Rank - 1)
+                    Dim commas = SyntaxFactory.TokenList(Enumerable.Repeat(SyntaxFactory.Token(SyntaxKind.CommaToken), commaCount))
+                    ranks.Add(SyntaxFactory.ArrayRankSpecifier(SyntaxFactory.Token(SyntaxKind.OpenParenToken), commas, SyntaxFactory.Token(SyntaxKind.CloseParenToken)))
+                    arrayType = TryCast(arrayType.ElementType, IArrayTypeSymbol)
+                End While
```

Comment:
nit: you're just digging into through `symbol.ElementType` here and above. Can we just combine into a single loop?